### DISCUSSION TITLE
Can the keyboard be flashed without mounting as a drive?

### DISCUSSION THREAD

**Discussion by jasonish** [KinesisCorporation/Adv360-Pro-ZMK#119] [Q&A] [answered]
I'm wonder if the Advantage 360 Pro can be flashed without copying the firmware to the virtual disk?  Something more like QMK's flash then entering boot mode on the keyboard with a shortcut.

Thanks.

**Comment by ReFil** [comment, marked as answer]
No, the usb drive bootloader is the only way to load new keymaps onto the Adv360 pro

**Reply by jasonish** [reply]
Thanks.  I was able to come up with a script to mount, copy as needed.

**Reply by garriguv** [reply]
@jasonish would you mind sharing a Gist of that script? 😃

**Reply by jasonish** [reply]
> @jasonish would you mind sharing a Gist of that script? smiley

Sure. It's Linux only though...

https://github.com/jasonish/Adv360-Pro-ZMK/blob/V2.0/flash.sh

**Reply by ReFil** [reply]
Looking at this I think it's using the unique chip IDs as part of the left/right identification stuff. That will probably need changing to work with other boards.

Looks really good though, it's awesome to see people coming up with these solutions and sharing them

**Reply by jasonish** [reply]
Yeah, I guess I comment on that in the script, but its worth noting externally to those who may not read it.

When the halves are not in bootloader mode, `lsusb` can identify which is the left and the right as the descriptions are different, however, once in bootloader mode I haven't seen a way to identify which half is which? Any thoughts?

**Reply by ReFil** [reply]
They run the same bootloader so there isn't really a way to differentiate unfortunately
